[J4] New module - Category Tags

korenevskiy opened the pull request:
### Actual result BEFORE applying this Pull Request
Previously, the display was very restrictive. Tag links opened the tag component. But the problem is that the Tag Component does not have a mode for displaying articles as a blog. At the same time, the display of the list of materials of one tag must be made additional layout.


### Expected result AFTER applying this Pull Request
The new module allows you to display tags in the form of a tree, This will allow you to use tags to display as a catalog of categories. Every developer faces a problem when an article needs to be placed in 2(3) different categories at once. For example, an article can simultaneously contain future news, as well as an article can contain information about the PHP language, as well as contain information about gifts. By opening the module tag link, a list of blog materials will be displayed.

While deleting the material tag, the material link remains the same. But if we moved an article from one category to another, the article link would change. Now the article link will remain the same, and tags can be used instead of categories.

![image](https://user-images.githubusercontent.com/6898474/189620007-b202589b-30b4-4a35-bd4b-2661cb365376.png)

UPDATED DESCRIPTION
_A known problem._
 After clicking on the tag links, the Category, Blog component stores this tag in the cache, and later, when opening any Category and Blog links, materials with the last open tag will be displayed.

_A quick solution to the problem._ 
In the link settings in the component menu, in the tag selection field, you must specify the selection of an empty tag. This will allow you to open links by resetting the tag filter in the cache.

_The required Solution to the problem._ 
It is required to make changes to the code of the Category and Blog component so that links without the selected tag reset the tag filter in the cache.

korenevskiy committed:
Update FieldsHelper.php

korenevskiy committed:
Merge branch '4.2-dev' into 4.2-dev

korenevskiy committed:
Update FieldsHelper.php

korenevskiy committed:
Update FieldsHelper.php

korenevskiy committed:
Update FieldsHelper.php

korenevskiy committed:
Merge branch '4.2-dev' into 4.2-dev

korenevskiy committed:
Update FieldsHelper.php

korenevskiy committed:
Merge branch '4.2-dev' into 4.2-dev

korenevskiy committed:
Merge branch '4.2-dev' into 4.2-dev

joomla-bot committed:
Merge tag 'psr12anchor' into psr12/merge/37765

joomla-bot committed:
Phase 1 convert BRANCH to PSR-12

joomla-bot committed:
Phase 2 convert BRANCH to PSR-12

joomla-bot committed:
Merge tag 'psr12final' into psr12/merge/37765

korenevskiy committed:
Merge branch '4.2-dev' into 4.2-dev

zero-24 committed:
update branch and set the correct version number for 4.3.0-dev

Hackwar committed:
Enabling packaging for 4.3-dev in drone (#38331)

bembelimen committed:
Merge pull request #38424 from joomla/4.2-dev

Updating 4.3-dev with changes from 4.2-dev

joomdonation committed:
[4.3] Replace Factory::getUser() by $this->getCurrentUser() in components models (#38090)

* Replace Factory::getUser() by $this->getCurrentUser()

richard67 committed:
Deprecate SqlsrvChangeItem for 5.0 (#38428)

obuisard committed:
Merge branch '4.2-dev' into upmerge-2022-08-22

# Conflicts:
#	administrator/language/en-GB/install.xml
#	administrator/language/en-GB/langmetadata.xml
#	administrator/manifests/files/joomla.xml
#	administrator/manifests/packages/pkg_en-GB.xml
#	api/language/en-GB/install.xml
#	api/language/en-GB/langmetadata.xml
#	installation/language/en-GB/langmetadata.xml
#	language/en-GB/install.xml
#	language/en-GB/langmetadata.xml
#	libraries/src/Version.php

obuisard committed:
Merge pull request #38557 from obuisard/upmerge-2022-08-22

Merge branch 4.2 into 4.3

Hackwar committed:
[4.3] Switching from Codeception to Cypress for tests (#38422)

laoneo committed:
[4.3] Activate all unit tests as default (#38421)

korenevskiy committed:
Merge branch '4.2-dev' into 4.2-dev

HLeithner committed:
Merge branch '4.2-dev' into 4.3-dev

korenevskiy committed:
Merge branch '4.2-dev' into 4.2-dev

heelc29 committed:
[4.3] Fix Tinymce and Codemirror version (#38614)

obuisard committed:
Merge branch '4.2-dev' into upmerge-2022-09-03

# Conflicts:
#	.gitignore
#	administrator/language/en-GB/install.xml
#	administrator/language/en-GB/langmetadata.xml
#	administrator/manifests/files/joomla.xml
#	administrator/manifests/packages/pkg_en-GB.xml
#	api/language/en-GB/install.xml
#	api/language/en-GB/langmetadata.xml
#	installation/language/en-GB/langmetadata.xml
#	language/en-GB/install.xml
#	language/en-GB/langmetadata.xml
#	libraries/src/Version.php

obuisard committed:
Merge pull request #38679 from obuisard/upmerge-2022-09-03

Upmerge 2022 09 03

sonvnn committed:
[4.3] Fix issue Deprecated: trim(): Passing null on PHP 8.1 (#38511)

Hackwar committed:
Switching Codeception error output to cypress output folder (#38712)

brianteeman committed:
tag alias (#38576)

brianteeman committed:
Admin Menu filter by parent (#38540)

obuisard committed:
Merge branch '4.2-dev' into upmerge-2022-09-09

obuisard committed:
Merge pull request #38733 from obuisard/upmerge-2022-09-09

Updating 4.3-dev with changes from 4.2-dev

Hackwar committed:
Changing name of install spec (#38736)

HLeithner committed:
Reorder cypress tests

webuniverse committed:
Update com_users.sys.ini (#38734)

If the user registration is not active, the menu item redirects to the login form. No message appears why the registration form is not displayed.

Co-authored-by: Allon Moritz <allon.moritz@digital-peak.com>

korenevskiy committed:
Add files via upload

korenevskiy committed:
Merge branch '4.2-dev' into j4_category_tags

korenevskiy committed:
Delete catalogCustom.php

brianteeman commented:
you dont need to include the russian language files

brianteeman commented:
and is the file translate.php part of this module?

korenevskiy committed:
Update default.php

korenevskiy committed:
Update _childs.php

korenevskiy committed:
Update cloud.php

korenevskiy committed:
Update mod_category_tags.php

korenevskiy committed:
Update CategoryTagsHelper.php

korenevskiy committed:
Update _childs.php

korenevskiy committed:
Update default.php

korenevskiy committed:
Update _childs.php

korenevskiy commented:
> and is the file translate.php part of this module?

No, but this file serves as a translation for testing. I would prefer to delete it after finding the errors.

korenevskiy committed:
Update mod_category_tags.php

korenevskiy committed:
Update cloud.php

korenevskiy committed:
Update mod_category_tags.php

korenevskiy committed:
Update cloud.php

korenevskiy committed:
Update _childs.php

korenevskiy committed:
Delete mod_category_tags.ini

korenevskiy committed:
Update CategoryTagsHelper.php

korenevskiy committed:
Merge branch '4.2-dev' into j4_category_tags

korenevskiy committed:
Update _childs.php

korenevskiy committed:
Add files via upload

korenevskiy committed:
Delete translate.php

korenevskiy committed:
Update _childs.php

korenevskiy committed:
Update cloud.php

korenevskiy committed:
Update mod_category_tags.php

korenevskiy committed:
Delete FieldsHelper.php

korenevskiy committed:
Create Helper

korenevskiy committed:
Delete Helper

korenevskiy committed:
Create FieldsHelper.php

korenevskiy committed:
Update _childs.php

korenevskiy committed:
Update cloud.php

korenevskiy committed:
Update mod_category_tags.php

korenevskiy committed:
Update CategoryTagsHelper.php

korenevskiy committed:
Update cloud.php

korenevskiy committed:
Update _childs.php

korenevskiy committed:
Update CategoryTagsHelper.php

korenevskiy committed:
Update mod_category_tags.php

korenevskiy committed:
Update _childs.php

korenevskiy committed:
Update mod_category_tags.php

korenevskiy committed:
Update CategoryTagsHelper.php

korenevskiy committed:
Update _childs.php

korenevskiy committed:
Update cloud.php

korenevskiy committed:
Update mod_category_tags.php

korenevskiy committed:
Update CategoryTagsHelper.php

korenevskiy committed:
Update mod_category_tags.php

korenevskiy committed:
Update cloud.php

korenevskiy committed:
Update mod_category_tags.php

korenevskiy committed:
Update cloud.php

korenevskiy committed:
Update _childs.php

korenevskiy committed:
Update CategoryTagsHelper.php

korenevskiy committed:
Update _childs.php

korenevskiy committed:
Update cloud.php

korenevskiy committed:
Update cloud.php

korenevskiy committed:
Update _childs.php

korenevskiy committed:
Update mod_category_tags.php

korenevskiy committed:
Update CategoryTagsHelper.php

korenevskiy committed:
Update _childs.php

korenevskiy committed:
Update cloud.php

korenevskiy committed:
Update mod_category_tags.php

korenevskiy committed:
Update _childs.php

korenevskiy committed:
Update cloud.php

korenevskiy committed:
Update _childs.php

korenevskiy committed:
Update _childs.php

korenevskiy committed:
Update mod_category_tags.php

korenevskiy committed:
Update _childs.php

korenevskiy committed:
Update cloud.php

korenevskiy committed:
Update _childs.php

korenevskiy committed:
Update cloud.php

korenevskiy committed:
Update mod_category_tags.php

korenevskiy committed:
Update cloud.php

korenevskiy committed:
Update _childs.php

korenevskiy committed:
Update mod_category_tags.php

korenevskiy committed:
Update _childs.php

korenevskiy committed:
Update cloud.php

korenevskiy committed:
Update cloud.php

korenevskiy committed:
Update _childs.php

korenevskiy committed:
Update CategoryTagsHelper.php

alikon commented:
if you look at https://ci.joomla.org/joomla/joomla-cms/57815/1/6 there is a detailed report on what drone complain about CS
![image](https://user-images.githubusercontent.com/181681/189709975-dee7912b-214c-4cfb-a736-a6fc5329a990.png)

hope that helps

imho shouldn't be a core but a 3dp module

korenevskiy commented:
@alikon  
Yes, I have been looking into this report from the very beginning. I have already removed 80% of the errors. 
But I can't fix the remaining errors. You will see for yourself the code and the error in the report. I do not understand that there is an error in an empty place. 
After that, I correct the errors as indicated in the report, and in the new report the requirement to remove the added space back. Ie, first the report asks to add a space, and in the next report asks to remove the space back. 

Maybe I don't understand something?

alikon committed:
fix for 1st only

26 | ERROR | [x] Expected 1 space(s) after closing parenthesis;
    |       |     found newline

brianteeman commented:
I would suggest not to worry for now about code style and first to get people to check what the module does and if it is something for core. _(there are lots of other style issues that drone didnt find)_

alikon commented:
joomla recently switched from own cosdestyle to PSR-12 https://www.php-fig.org/psr/psr-12/
so maybe this is the main source of confusion
i've made a silly pr to your branch  to fix  only the 1st drone report 
check it maybe it will let you to reduce the cs complains

korenevskiy committed:
Merge pull request #2 from alikon/patch-11

fix for 1st only

korenevskiy committed:
Update mod_category_tags.php

korenevskiy committed:
Update CategoryTagsHelper.php

korenevskiy committed:
Update _childs.php

korenevskiy committed:
Update cloud.php

korenevskiy committed:
Update CategoryTagsHelper.php

korenevskiy committed:
Update _childs.php

korenevskiy committed:
Update mod_category_tags.php

korenevskiy commented:
> so maybe this is the main source of confusion

@alikon 
You're right, the problem was in the curly brace after the condition. But in the error report, it is NOT stated that the problem is curly brace. The report just says about the brace. And since the curly brace is on another line, that implies a parenthesis, as well as the error is indicated in the absence of a space. The error in the report does not indicate the cause of the error at all.

**Thank you for your help, now I will know about the new code format.**

korenevskiy committed:
Update _childs.php

korenevskiy committed:
Update _childs.php

korenevskiy committed:
Update _childs.php

korenevskiy committed:
Update mod_category_tags.php

korenevskiy committed:
Update default.php

korenevskiy commented:
`modules/mod_category_tags/tmpl/_childs.php (line 1)`
@alikon 
I have fixed all the errors, there is one more error that I cannot remove. Please help me fix it. The parser swears that the line break "\r\n" is used instead of "\n". No matter how much I would try to fix the transfer, the error remains in the code.

korenevskiy committed:
Update _childs.php

richard67 commented:
@korenevskiy Regarding naming of files and variables: The plural of "child" in English language is "children", not "childs".

richard67 commented:
> `modules/mod_category_tags/tmpl/_childs.php (line 1)` @alikon I have fixed all the errors, there is one more error that I cannot remove. Please help me fix it. The parser swears that the line break "\r\n" is used instead of "\n". No matter how much I would try to fix the transfer, the error remains in the code.

@korenevskiy You could use a good editor like notepad++ which can show you what kind of line feeds are in the file (Unix like it should be, or Windows like the PHP CS check complains about).

korenevskiy committed:
Update mod_category_tags.xml

korenevskiy committed:
Update mod_category_tags.ini

korenevskiy committed:
Update mod_category_tags.xml

korenevskiy committed:
Update _childs.php

korenevskiy committed:
Update _childs.php

korenevskiy committed:
Update _childs.php

korenevskiy committed:
Update mod_category_tags.xml

korenevskiy committed:
Update _childs.php

korenevskiy committed:
Update _childs.php

korenevskiy committed:
Update cloud.php

Kostelano commented on modules/mod_category_tags/mod_category_tags.php:
```diff
@@ -0,0 +1,70 @@
+ * @package     Joomla.Site
+ * @subpackage  mod_category_tags
+ *
+ * @copyright   (C) 2013 Open Source Matters, Inc. <https://www.joomla.org>
```
```suggestion
 * @copyright   (C) 2022 Open Source Matters, Inc. <https://www.joomla.org>
```

Kostelano commented on modules/mod_category_tags/mod_category_tags.xml:
```diff
@@ -0,0 +1,252 @@
+<extension type="module" client="site" method="upgrade">
+	<name>mod_category_tags</name>
+	<author>Joomla! Project</author>
+	<creationDate>2022-05-09</creationDate>
+	<copyright>(C) 2013 Open Source Matters, Inc.</copyright>
+	<license>GNU General Public License version 2 or later; see LICENSE.txt</license>
+	<authorEmail>admin@joomla.org</authorEmail>
+	<authorUrl>www.joomla.org</authorUrl>
+	<version>4.0.0</version>
```
```suggestion
	<creationDate>2022-05-09</creationDate>
	<copyright>(C) 2022 Open Source Matters, Inc.</copyright>
	<license>GNU General Public License version 2 or later; see LICENSE.txt</license>
	<authorEmail>admin@joomla.org</authorEmail>
	<authorUrl>www.joomla.org</authorUrl>
	<version>__DEPLOY_VERSION__</version>
```

korenevskiy replied:
Why do we need to write the "creationDate" not in SQL format?

Kostelano replied:
[Joomla recently switched to this date format](https://github.com/joomla/joomla-cms/pull/27162). You can check in any module.

korenevskiy replied:
> [Joomla recently switched to this date format](https://github.com/joomla/joomla-cms/pull/27162). You can check in any module.

This is a big mistake of all modules. An XML file is a data file just like SQL is designed for data. The XML file is not a text document. We store data in SQL in ISO format. Maybe we will store data in SQL in the "Nov 2022" format?

Thank you so much. I have made all your corrections except for the CreateionDate tag.

Kostelano replied:
I proposed a correction based on the previously adopted PR (for compliance). But most likely you are right and this is a mistake. There is already at least an https://github.com/joomla/joomla-cms/issues/38552 on this topic.

Kostelano replied:
I have edited the change request. The date will need to be discussed with the release leaders.

korenevskiy replied:
> There is already at least an #38552 on this topic.

Thank you, I indicated there in the message that XML is a data format.

Kostelano commented on language/en-GB/mod_category_tags.ini:
```diff
@@ -0,0 +1,43 @@
+; Joomla! Project
+; (C) 2013 Open Source Matters, Inc. <https://www.joomla.org>
+; License GNU General Public License version 2 or later; see LICENSE.txt
+; Note : All ini files need to be saved as UTF-8
+
+MOD_CATEGORY_TAGS="Category - tags"
+MOD_CATEGORY_TAGS_FIELD_ALL_TIME="All time"
+MOD_CATEGORY_TAGS_FIELD_DISPLAY_COUNT_LABEL="Display Number of Items"
+MOD_CATEGORY_TAGS_FIELD_LAST_DAY="Last day"
+MOD_CATEGORY_TAGS_FIELD_LAST_HOUR="Last hour"
+MOD_CATEGORY_TAGS_FIELD_LAST_MONTH="Last month"
+MOD_CATEGORY_TAGS_FIELD_LAST_WEEK="Last week"
+MOD_CATEGORY_TAGS_FIELD_LAST_YEAR="Last year"
+MOD_CATEGORY_TAGS_FIELD_MAXSIZE_DESC="The maximum font size used for the tags, proportional to the site's default font size (eg &quot;2&quot; means 200% of the default size)."
+MOD_CATEGORY_TAGS_FIELD_MAXSIZE_LABEL="Maximum Font Size"
+MOD_CATEGORY_TAGS_FIELD_MINSIZE_DESC="The minimum font size used for the tags, proportional to the site's default font size (eg &quot;2&quot; means 200% of the default size)."
+MOD_CATEGORY_TAGS_FIELD_MINSIZE_LABEL="Minimum Font Size"
+MOD_CATEGORY_TAGS_FIELD_NO_RESULTS_LABEL="Show &quot;No results&quot; text"
+MOD_CATEGORY_TAGS_FIELD_NO_RESULTS_TEXT = "No results message"
+MOD_CATEGORY_TAGS_NO_ITEMS_FOUND="No Tags found."
+MOD_CATEGORY_TAGS_FIELD_ORDER_VALUE_COUNT="Number of Items"
+MOD_CATEGORY_TAGS_FIELD_ORDER_VALUE_LABEL="Order"
+MOD_CATEGORY_TAGS_FIELD_ORDER_VALUE_RANDOM="Random"
+MOD_CATEGORY_TAGS_FIELD_ORDER_VALUE_TITLE="Title"
+MOD_CATEGORY_TAGS_FIELD_TIMEFRAME_LABEL="Time Period"
+MOD_CATEGORY_TAGS_FIELD_DISPLAY_TREE_LABEL ="Show tree"
```
```suggestion
; (C) 2022 Open Source Matters, Inc. <https://www.joomla.org>
; License GNU General Public License version 2 or later; see LICENSE.txt
; Note : All ini files need to be saved as UTF-8

MOD_CATEGORY_TAGS="Category - Tags"
MOD_CATEGORY_TAGS_FIELD_ALL_TIME="All time"
MOD_CATEGORY_TAGS_FIELD_DISPLAY_COUNT_LABEL="Display Number of Items"
MOD_CATEGORY_TAGS_FIELD_LAST_DAY="Last day"
MOD_CATEGORY_TAGS_FIELD_LAST_HOUR="Last hour"
MOD_CATEGORY_TAGS_FIELD_LAST_MONTH="Last month"
MOD_CATEGORY_TAGS_FIELD_LAST_WEEK="Last week"
MOD_CATEGORY_TAGS_FIELD_LAST_YEAR="Last year"
MOD_CATEGORY_TAGS_FIELD_MAXSIZE_DESC="The maximum font size used for the tags, proportional to the site's default font size (eg \"2\" means 200% of the default size)."
MOD_CATEGORY_TAGS_FIELD_MAXSIZE_LABEL="Maximum Font Size"
MOD_CATEGORY_TAGS_FIELD_MINSIZE_DESC="The minimum font size used for the tags, proportional to the site's default font size (eg \"2\" means 200% of the default size)."
MOD_CATEGORY_TAGS_FIELD_MINSIZE_LABEL="Minimum Font Size"
MOD_CATEGORY_TAGS_FIELD_NO_RESULTS_LABEL="Show \"No results\" text"
MOD_CATEGORY_TAGS_FIELD_NO_RESULTS_TEXT="No results message"
MOD_CATEGORY_TAGS_NO_ITEMS_FOUND="No Tags found."
MOD_CATEGORY_TAGS_FIELD_ORDER_VALUE_COUNT="Number of Items"
MOD_CATEGORY_TAGS_FIELD_ORDER_VALUE_LABEL="Order"
MOD_CATEGORY_TAGS_FIELD_ORDER_VALUE_RANDOM="Random"
MOD_CATEGORY_TAGS_FIELD_ORDER_VALUE_TITLE="Title"
MOD_CATEGORY_TAGS_FIELD_TIMEFRAME_LABEL="Time Period"
MOD_CATEGORY_TAGS_FIELD_DISPLAY_TREE_LABEL="Show tree"
```

Kostelano commented on language/en-GB/mod_category_tags.sys.ini:
```diff
@@ -0,0 +1,9 @@
+; Joomla! Project
+; (C) 2013 Open Source Matters, Inc. <https://www.joomla.org>
+; License GNU General Public License version 2 or later; see LICENSE.txt
+; Note : All ini files need to be saved as UTF-8
+
+MOD_CATEGORY_TAGS="Category - tags"
```
```suggestion
; (C) 2022 Open Source Matters, Inc. <https://www.joomla.org>
; License GNU General Public License version 2 or later; see LICENSE.txt
; Note : All ini files need to be saved as UTF-8

MOD_CATEGORY_TAGS="Category - Tags"
```

Kostelano commented on modules/mod_category_tags/src/Helper/CategoryTagsHelper.php:
```diff
@@ -0,0 +1,198 @@
+ * @package     Joomla.Site
+ * @subpackage  mod_category_tags
+ *
+ * @copyright   (C) 2013 Open Source Matters, Inc. <https://www.joomla.org>
```
```suggestion
 * @copyright   (C) 2022 Open Source Matters, Inc. <https://www.joomla.org>
```

Kostelano commented on modules/mod_category_tags/tmpl/_childs.php:
```diff
@@ -0,0 +1,67 @@
+ * @package     Joomla.Site
+ * @subpackage  mod_category_tags
+ *
+ * @copyright   (C) 2013 Open Source Matters, Inc. <https://www.joomla.org>
```
```suggestion
 * @copyright   (C) 2022 Open Source Matters, Inc. <https://www.joomla.org>

```

Kostelano commented on modules/mod_category_tags/tmpl/cloud.php:
```diff
@@ -0,0 +1,63 @@
+ * @package     Joomla.Site
+ * @subpackage  mod_category_tags
+ *
+ * @copyright   (C) 2013 Open Source Matters, Inc. <https://www.joomla.org>
```
```suggestion
 * @copyright   (C) 2022 Open Source Matters, Inc. <https://www.joomla.org>
```

Kostelano commented on modules/mod_category_tags/tmpl/default.php:
```diff
@@ -0,0 +1,31 @@
+ * @package     Joomla.Site
+ * @subpackage  mod_category_tags
+ *
+ * @copyright   (C) 2013 Open Source Matters, Inc. <https://www.joomla.org>
```
```suggestion
 * @copyright   (C) 2022 Open Source Matters, Inc. <https://www.joomla.org>
```

Kostelano commented on modules/mod_category_tags/language/en-GB/mod_category_tags.ini:
```diff
@@ -0,0 +1,40 @@
+; Joomla! Project
```
This file and the "language" directory must be deleted, no localization files are stored in the modules directory.

korenevskiy committed:
Update info date /mod_category_tags.php

Co-authored-by: Konstantin Kolos <k0nstantinkolos88@gmail.com>

korenevskiy committed:
Update language/en-GB/mod_category_tags.ini

Co-authored-by: Konstantin Kolos <k0nstantinkolos88@gmail.com>

korenevskiy committed:
Update language/en-GB/mod_category_tags.sys.ini

Co-authored-by: Konstantin Kolos <k0nstantinkolos88@gmail.com>

korenevskiy committed:
Update modules/mod_category_tags/src/Helper/CategoryTagsHelper.php

Co-authored-by: Konstantin Kolos <k0nstantinkolos88@gmail.com>

korenevskiy committed:
Update modules/mod_category_tags/tmpl/cloud.php

Co-authored-by: Konstantin Kolos <k0nstantinkolos88@gmail.com>

korenevskiy committed:
Delete mod_category_tags.ini

korenevskiy committed:
Update modules/mod_category_tags/tmpl/_childs.php

Co-authored-by: Konstantin Kolos <k0nstantinkolos88@gmail.com>

korenevskiy committed:
Update modules/mod_category_tags/tmpl/default.php

Co-authored-by: Konstantin Kolos <k0nstantinkolos88@gmail.com>

brianteeman commented on modules/mod_category_tags/mod_category_tags.xml:
```diff
@@ -0,0 +1,252 @@
+					label="MOD_CATEGORY_TAGS_FIELD_CATEGORY_LAYOUT_LABEL"
+					default="blog"
+					validate="options"
+				>
```
Please fix the indentation here and in all similar areas.

brianteeman commented on modules/mod_category_tags/mod_category_tags.xml:
```diff
@@ -0,0 +1,252 @@
+			<fieldset name="basic">
+				
+				<field
+					type="translate"/>
```
???

brianteeman commented on modules/mod_category_tags/mod_category_tags.xml:
```diff
@@ -0,0 +1,252 @@
+
+
+				<field
+					name="tree_display"
+					type="radio"
+					layout="joomla.form.field.radio.switcher"
+					label="MOD_CATEGORY_TAGS_FIELD_DISPLAY_TREE_LABEL"
```
name
type
label
layout

please fix this ordering here and in all similar areas

korenevskiy replied:
in new PR changed

brianteeman commented on modules/mod_category_tags/mod_category_tags.xml:
```diff
@@ -0,0 +1,252 @@
+					layout="joomla.form.field.radio.switcher"
+					label="MOD_CATEGORY_TAGS_FIELD_DISPLAY_TITLE_LABEL"
+					default="1"
+					filter="integer"
```
filter="boolean"

korenevskiy replied:
in new PR changed

Kostelano commented:
By the way, replace the branch with 4.3.

korenevskiy committed:
Update mod_category_tags.xml

korenevskiy committed:
Merge branch '4.3-dev' into j4_category_tags

korenevskiy committed:
Merge branch 'joomla:4.2-dev' into j4_category_tags

korenevskiy commented:
Dear colleagues. I have created a new PR for branch 4.3. I am not able to transfer PR to a new branch. I decided to create anew. Here's a new PR https://github.com/joomla/joomla-cms/pull/38752

@brianteeman , @Kostelano  , @chmst  , @alikon  ,  @richard67  , @laoneo  , @rdeutz  , @Hackwar  , @wilsonge  , @zero-24  , @bembelimen , @joomdonation  , @obuisard  , @HLeithner  , @heelc29  , @sonvnn @webuniverse
